docs/source/deployment/aws_ec2.rst

```diff
@@ -0,0 +1,370 @@
+
+
+BentoML has great support for AWS EC2. You can deploy, update and delete
+deployment with single command, and customize deployment to fit your needs with parameters
+such as `instance type`,`scaling capacities`
+
+To deploy BentoService to AWS EC2, use `bentoml lambda deploy` command.
+Deployment name and bento service name:version tag is required.
+
+.. code-block:: bash
+
+    > bentoml ec2 deploy my-first-ec2-deployment -b IrisClassifier:20200121141808_FE78B5
+
+    Deploying EC2 deployment -[2020-01-21 14:37:16,838] INFO - Building project
+    [2020-01-21 14:38:52,826] INFO - Containerzing service
+    [2020-01-21 14:39:18,834] INFO - Deploying
+    [2020-01-21 14:40:09,265] INFO - ApplyDeployment (my-first-ec2-deployment, namespace dev) succeeded
+    Successfully created AWS EC2 deployment my-first-ec2-deployment
+    {
+    "namespace": "dev",
+    "name": "my-first-ec2-deployment",
+    "spec": {
+        "bentoName": "IrisClassifier",
+        "bentoVersion": "20200121141808_FE78B5",
+        "operator": "AWS_EC2",
+        "awsEc2OperatorConfig": {
+        "region": "ap-south-1",
+        "instanceType": "t2.micro",
+        "amiId": "/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2",
+        "autoscaleMinCapacity": 1,
+        "autoscaleDesiredCapacity": 1,
+        "autoscaleMaxCapacity": 1
+        }
+    },
+    "state": {},
+    "createdAt": "2020-10-24T06:52:18.003580Z",
+    "lastUpdatedAt": "2020-10-24T06:52:18.003626Z"
+    }
+
+
+
+BentoML helps you manage the entire process of deploying your BentoService bundle to EC2.
+Verify the deployed resources with AWS CLI tool:
+
+.. code-block:: bash
+
+    > aws cloudformation describe-stacks
+
+    {
+        "Stacks": [
+            {
+                "StackId": "arn:aws:cloudformation:us-west-2:192023623294:stack/dev-my-first-lambda-deployment/dd2a7cf0-3c9e-11ea-8654-02f6ffa9fe66",
+                "StackName": "dev-my-first-lambda-deployment",
+                "ChangeSetId": "arn:aws:cloudformation:us-west-2:192023623294:changeSet/samcli-deploy1579646359/f9c876ca-ade0-4623-93e9-870ef6e7e1b5",
+                "CreationTime": "2020-01-21T22:39:20.156Z",
+                "LastUpdatedTime": "2020-01-21T22:39:25.602Z",
+                "RollbackConfiguration": {},
+                "StackStatus": "CREATE_COMPLETE",
+                "DisableRollback": false,
+                "NotificationARNs": [],
+                "Capabilities": [
+                    "CAPABILITY_IAM"
+                ],
+                "Outputs": [
+                    {
+                        "OutputKey": "S3Bucket",
+                        "OutputValue": "btml-dev-my-first-lambda-deployment-a4a791",
+                        "Description": "S3 Bucket for saving artifacts and lambda bundle"
+                    },
+                    {
+                        "OutputKey": "Url",
+                        "OutputValue": "https://ps6f0sizt8.execute-api.us-west-2.amazonaws.com/",
```

Comment:
Will this URL be changed after a `bentoml ec2 update my-first-ec2-deployment -b IrisClassifier:another_version` call?

Reply:
No it wont change,because only scaling-group replace instances,balancer stays same.
This will only change if balancer is replaced which doesn't happen in deployment cycle of bentos.